src/AJT/Toggl/ReportsClient.php

```diff
@@ -23,38 +20,25 @@ class ReportsClient extends Client
      *
      * See https://www.toggl.com/public/api#api_token for more information on the api token
      *
-     * @param array|Collection $config Configuration data
-     *
+     * @param array $config Configuration data
      * @return ReportsClient
+     * @throws \Exception
      */
-    public static function factory($config = array())
+    public static function factory($config = [])
     {
-        $default = array(
-            'base_url' => 'https://www.toggl.com/reports/api/{apiVersion}',
-            'debug' => false,
-            'apiVersion' => 'v2'
-        );
-        $required = array('api_key', 'base_url', 'apiVersion');
-        $config = Collection::fromConfig($config, $default, $required);
-
-        $serviceDescriptionFile = __DIR__ . '/reporting_' . $config->get('apiVersion') . '.json';
-        $description = ServiceDescription::factory($serviceDescriptionFile);
 
-        $client = new self($config->get('base_url'), $config);
+        $clientConfig = self::getClientConfig($config);
 
-        $client->setDescription($description);
+        $guzzleClient = new Client($clientConfig);
 
-        $client->setDefaultHeaders(array(
-            "Content-type" => "application/json",
-        ));
+        if (isset($config['apiVersion']) && $config['apiVersion'] !== 'v2') {
+            throw new \Exception('Only v8 is supported at this time');
```

Comment:
Ah yes, copy paste error. I'm not sure why to keep 2 clients who do almost the same except for the different definition file. But i'm not gonna change that ;)